weather.py

```diff
@@ -1,72 +1,143 @@
 #!/usr/bin/env python3
 
+from dataclasses import dataclass
+from datetime import datetime
+import dacite
 import json
 import requests
+import sys
 
 
 # get_locations() accepts a url as a parameter and returns a dictionary
 # of locations.
 def get_locations(url):
-    # Make the http requests to the url.
     resp = requests.get(url)
 
-    # If the resonse status code is not 200, then return nothing.
-    # TODO: We might want log an error here.
-    # Also worth consdering the behaviour here if the url is unreachable, and
-    # how we handle different status codes.
     if resp.status_code != 200:
         print(resp.status_code)
         return None
 
-    # Initialise an empty dictionary.
     d = {}
 
-    # resp.text is a json object that is a list. And when we pass it into
-    # json.loads() we get a list.
-    # Here we are literally saying: for each item in the list that we have
-    # created from the json object, d's key is the value of 'LocalizedName',
-    # and d's value is the value of 'ID'. This allows us to get the ID
-    # from the human readable name.
-    # This is creating a dictionary, which is easier to get values from, rather
-    # than iterating over a list.
     for item in json.loads(resp.text):
         d[item['LocalizedName']] = item['ID']
 
     return d
 
 
+def make_request(url):
+#    print("making_request()")
```

Comment:
Please pay attention to indentation. The `#` should be indented the same as the code within the function scope.

Reply:
OK. Fixed.